### PR TITLE
fitboot: boot a bzImage from a specialized FIT image 

### PR DESCRIPTION
**Pull request by chengchiehhuang**
This fitboot will look for a "bzImage configuration" in the FIT image and boot them.

The convention will be if default config in FIT image is called conf@1, fitboot will then look for conf_bz@1 in FIT config section.

### CODE REVIEW

File: pkg/boot/fit/fit.go [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+// Load LinuxImage to memory
+func loadLinuxImage(i *boot.LinuxImage, verbose bool) error {
+	return i.Load(verbose)
+}
+
+// provide chance to mock in test
+var loadImage = loadLinuxImage
+var kexecReboot = kexec.Reboot
+
+// Load image and reboot
```
Comment (rminnich) [inline pkg/boot/fit/fit.go:75]: Please change to
// Load loads an image and reboots

File: pkg/boot/fit/fit.go [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+	} else {
+		v("Not trying to boot since this is a dry run")
+	}
+
+	return nil
+}
+
+// Loads configuration from FIT Image.
+// Returns <kernel_name>, <ramdisk_name>, error
+func (i *Image) LoadBzConfig(verbose bool) (string, string, error) {
```
Comment (rminnich) [inline pkg/boot/fit/fit.go:118]: Please change to 
LoadBzConfig loads a configuration from a FIT image

Also you can run 
go vet .
and
golint .
in the package and command directory to get useful warnings.

**Review by rminnich — COMMENTED** [review]
Just a few minor comments. This is just about ready.

File: pkg/boot/fit/fit.go (hunk 2 above)
Comment (chengchiehhuang) [reply pkg/boot/fit/fit.go:118]: Fixed.

Thanks for the go vet and golint hints.

File: pkg/boot/fit/fit.go (hunk 1 above)
Comment (chengchiehhuang) [reply pkg/boot/fit/fit.go]: Fixed

**Comment by chengchiehhuang** [comment]
Hi Ron,
PTAL. I also set -rsdp default to false, because we also have a fixrsdp command where we copy RSDP to EBDA.

File: pkg/boot/fit/testdata/create_dummy_fit_image.sh [hunk 3]
```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
```
Comment (rminnich) [inline pkg/boot/fit/testdata/create_dummy_fit_image.sh:1]: We don't use bash script in u-root as it creates too many dependencies for systems that only have u-root in source mode. Could you redo this as a Go function, that does not call Unix commands (i.e. there should be no exec.Command) and just have it create a temporary file? Or, further, have a pre-declared []byte in Go source that is the file? How big is this file when it is created?

**Comment by chengchiehhuang** [comment]
I changed the script to a README file. Please take another look. Thanks.

**Review by rminnich — APPROVED** [review]

**Pull request merged**